### ISSUE TITLE
Blueprint issues

### ISSUE THREAD

**Issue by pvliss** [jhipster/generator-jhipster#9232]
##### **Overview of the issue**

So I am trying to add mocha unit tests for the Kotlin blueprint and I have stumbled upon the following problems while working with the `entity` generator:
1. Options are not propagated to the blueprint sub-generator. So for example the test `with angular suffix` of the`entity-spec.js` is failing as the `angular-suffix` option is passed to the blueprint.
2. Defining blueprint as an option to the runner is not being picked up by `entity/index.js`. The workaround for this is to define a `.yo-rc.json` file that also includes the blueprint.

##### **Motivation for or Use Case**
Create unit tests for the Kotlin blueprint with minimal maintenance effort.

##### **Reproduce the error**
Use code from the following branch: https://github.com/pvliss/jhipster-kotlin/tree/mocha-tests
Run the following:
```bash
npm install
npm link generator-jhipster
npm test
```

##### **Suggest a Fix**

1. Pass the options to the blueprint, i.e:
```js
            useBlueprint = this.composeBlueprint(blueprint, 'entity', {
                ...this.options,
                'skip-install': this.options['skip-install'],
                'from-cli': this.options['from-cli'],
                force: this.options.force,
                arguments: [this.context.name]
            });

```
**NOTE:** Some options are obviously duplicated and could be removed
2. The same as with [client generator](https://github.com/jhipster/generator-jhipster/blob/5ff10fc5b7c156cbddaee253c74bd6b96a19a364/generators/client/index.js#L65) which already supports loading the blueprint from options as well. i.e change to:
```js
        const blueprint = this.options.blueprint || this.configOptions.blueprint || this.config.get('blueprint');
```

Final note, is that I think that these should be applied to all generators that support blueprints not just the entity and entity-server.

##### **JHipster Version(s)**
Current master

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by pascalgrimaud** [comment]
@pvliss : for your information:
- https://github.com/jhipster/generator-jhipster-blueprint/issues/10#issuecomment-431983686
- https://github.com/jhipster/generator-jhipster-blueprint/pull/13

In fact, it seems to be difficult to write `npm test` for entity sub generator.
Our current tests for entities are bad:
- https://github.com/jhipster/generator-jhipster/blob/master/test/entity.spec.js

**Comment by pvliss** [comment]
Thanks for the extra info @pascalgrimaud!!!

So it seems that others are having similar issues as well. I actually got them working for the Kotlin blueprint. See here: https://github.com/pvliss/jhipster-kotlin/tree/mocha-tests/test.
Of course I executed them locally after making the changes suggested in my original comment

> In fact, it seems to be difficult to write `npm test` for entity sub generator.

Well in addition the problems already discussed I see that currently you can only test using the entity sub-generator and not with the entity-server or entity-client. Is that what you mean? By the way the same applies for the server as more tests are written against the app generator and just one for the server.

>Our current tests for entities are bad:
    * https://github.com/jhipster/generator-jhipster/blob/master/test/entity.spec.js

By bad you mean incomplete or the fact that there are mixed server and client and should be separated or the fact that they use files?

In any case I will do some PRs to help rectify some these issues.

**Comment by pascalgrimaud** [comment]
> By bad you mean incomplete or the fact that there are mixed server and client and should be separated or the fact that they use files?

We didn't answer the questions, as there is an infinite loop when using the entity sub generator. That's why we focus our tests with Travis/Azure for entities.

**Comment by pvliss** [comment]
Thanks for the extra info @pascalgrimaud. I kind of figured it out myself yesterday the hard way, as I was working on some more improvements mainly regarding the first issue mentioned here. Sadly I did not have time to deal with it today at all but will resume soon as I think these issues are critical for the Kotlin blueprint and other blueprints as well.

**Comment by pvliss** [comment]
Closed by  #9248 and #9266. Thanks @murdos  and @pascalgrimaud
